core/src/views/scrollview.rs

```diff
@@ -107,10 +107,15 @@ impl<L: Lens<Target = ScrollData>> ScrollView<L> {
         VStack::new(cx, content)
             .class("scroll_content")
             .bind(data.clone(), |handle, data| {
-                let data = data.get(handle.cx);
-                let left = ((data.child_x - data.parent_x) * data.scroll_x).round();
-                let top = ((data.child_y - data.parent_y) * data.scroll_y).round();
-                handle.left(Units::Pixels(-left)).top(Units::Pixels(-top));
+                let dpi_factor = handle.cx.style().dpi_factor;
+                if dpi_factor > 0.0 {
+                    let data = data.get(handle.cx);
+                    let left = ((data.child_x - data.parent_x) * data.scroll_x).round()
+                        / handle.cx.style().dpi_factor as f32;
+                    let top = ((data.child_y - data.parent_y) * data.scroll_y).round()
+                        / handle.cx.style().dpi_factor as f32;
+                    handle.left(Units::Pixels(-left.abs())).top(Units::Pixels(-top.abs()));
+                }
```

Comment:
What does this fix?

Reply:
The cache contains values in physical coordinates but the style properties expect logical. This fixes an bug where extra space gets put at the bottom of a scrollable area.